site/en/blog/soft-navigations-experiment/index.md

```diff
@@ -0,0 +1,258 @@
+
+The easiest way to take account of all the above nuances, is to use the [`web-vitals`](https://github.com/GoogleChrome/web-vitals) JavaScript library which has [experimental support for soft navigations](https://github.com/GoogleChrome/web-vitals/tree/soft-navs#report-metrics-for-soft-navigations-experimental) in a separate [`soft-navs branch`](https://github.com/GoogleChrome/web-vitals/tree/soft-navs) (which is also available on [npm](https://www.npmjs.com/package/web-vitals/v/soft-navs) and [unpkg](https://unpkg.com/web-vitals@soft-navs/dist/web-vitals.js?module)). This can be measured in the following way (replacing `doTraditionalProcessing` and `doSoftNavProcessing` as appropriate):
+
+```js
+import {
+  onCLS,
+  onFID,
+  onLCP,
+} from 'https://unpkg.com/web-vitals@soft-navs/dist/web-vitals.js?module';
+
+onTTFB(doTraditionalProcessing);
+onFCP(doTraditionalProcessing);
+onLCP(doTraditionalProcessing);
+onCLS(doTraditionalProcessing);
+onFID(doTraditionalProcessing);
+onINP(doTraditionalProcessing);
+
+onTTFB(doSoftNavProcessing, {reportSoftNavs: true});
+onFCP(doSoftNavProcessing, {reportSoftNavs: true});
+onLCP(doSoftNavProcessing, {reportSoftNavs: true});
+onCLS(doSoftNavProcessing, {reportSoftNavs: true});
+//onFID(doSoftNavProcessing, {reportSoftNavs: true});
+onINP(doSoftNavProcessing, {reportSoftNavs: true});
+```
+
+Ensure the metrics are reported against the correct URL [as noted above](#reporting-the-metrics-against-the-appropriate-url).
+
+The `web-vitals` library currently reports the following metrics for soft navigations:
+
+<table>
+  <thead>
+    <tr>
+      <th>Metric</th>
+      <th>Details</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>TTFB</td>
+      <td>Reported as 0.</td>
+    </tr>
+    <tr>
+      <td>FCP</td>
+      <td>The time of the next contentful paint, relevant to the soft navigation start time. Existing paints present from the previous navigation are not considered. Therefore, the FCP will be >= 0.</td>
+    </tr>
+    <tr>
+      <td>LCP</td>
+      <td>The time of the next largest contentful paint, relevant to the soft navigation start time. Existing paints present from the previous navigation are not considered. Therefore, the LCP will be >= 0. As usual, this will be reported upon an interaction, or when the page is backgrounded, as only then can the LCP be finalized.</td>
+    </tr>
+    <tr>
+      <td>CLS</td>
+      <td>The largest window of shifts between the navigation times. As usual, this when the page is backgrounded as only then can the CLS be finalized. A 0 value is reported.</td>
+    </tr>
+    <tr>
+      <td>FID</td>
+      <td>FID is not currently reported by the <code>web-vitals</code> library.</td>
+    </tr>
+    <tr>
+      <td>INP</td>
+      <td>The INP between the navigation times. As usual this will be reported upon an interaction, or when the page is backgrounded as only then can the INP be finalized. A 0 value is not reported.</td>
```

Comment:
A bit of an aside, but have you tested reporting of INP and the measurement of the interaction that led to the navigation itself?

For example, its common for next-FCP == INP --> does that report to the previous navigation or the next navigation?

Reply:
I do what event timing tells me 😁 Including the `navigationId` on that.

Reply:
(I filed an issue)